cpp/src/arrow/parquet/reader.cc

```diff
@@ -155,11 +159,53 @@ FlatColumnReader::Impl::Impl(MemoryPool* pool, const ::parquet::ColumnDescriptor
   NextRowGroup();
 }
 
+#define CONVERT_PHYSICAL_TYPE_COPY_DATA(InType, OutType)                    \
+  template <>                                                               \
+  Status FlatColumnReader::Impl::ConvertPhysicalType<InType, OutType>(      \
+      const InType* in_ptr, int64_t length, const OutType** out_ptr) {      \
+    RETURN_NOT_OK(values_builder_buffer_.Resize(length * sizeof(OutType))); \
+    OutType* mutable_out_ptr =                                              \
+        reinterpret_cast<OutType*>(values_builder_buffer_.mutable_data());  \
+    std::copy(in_ptr, in_ptr + length, mutable_out_ptr);                    \
+    *out_ptr = mutable_out_ptr;                                             \
+    return Status::OK();                                                    \
+  }
+
+#define CONVERT_PHYSICAL_TYPE_COPY_PTR(InType, OutType)                \
+  template <>                                                          \
+  Status FlatColumnReader::Impl::ConvertPhysicalType<InType, OutType>( \
+      const InType* in_ptr, int64_t length, const OutType** out_ptr) { \
+    *out_ptr = reinterpret_cast<const OutType*>(in_ptr);               \
+    return Status::OK();                                               \
+  }
+
+CONVERT_PHYSICAL_TYPE_COPY_PTR(bool, uint8_t)
+CONVERT_PHYSICAL_TYPE_COPY_DATA(int32_t, uint8_t)
+CONVERT_PHYSICAL_TYPE_COPY_DATA(int32_t, int8_t)
+CONVERT_PHYSICAL_TYPE_COPY_DATA(int32_t, uint16_t)
+CONVERT_PHYSICAL_TYPE_COPY_DATA(int32_t, int16_t)
+CONVERT_PHYSICAL_TYPE_COPY_PTR(int32_t, uint32_t)
+CONVERT_PHYSICAL_TYPE_COPY_PTR(int32_t, int32_t)
+CONVERT_PHYSICAL_TYPE_COPY_PTR(int64_t, uint64_t)
```

Comment:
I understand now that this is effectively "encoding" unsigned ints in a signed integer. Per above we should find out what other reader implementations are expecting (cast versus reinterpreting the bytes)